src/ai/susi/server/api/cms/FeedbackSkillService.java

```diff
@@ -0,0 +1,313 @@
+                                    skillName.put(i, feedbackObject);
+                                    alreadyByUser = true;
+                                    feedbackUpdated = true;
+                                    
+                                    // Update the skillRating.json file that contains overview of the ratings.
+                                    updateSkillRatingsJSON(call);
+                                    break;
+                                }
+                            }
+
+                            if (!alreadyByUser) {
+                                addToSkillRatingJSON(call);
+                            }
+                        }
+                        
+                        Timestamp timestamp = new Timestamp(System.currentTimeMillis());
+                        feedbackObject.put("email", email);
+                        feedbackObject.put("feedback", skill_feedback);
+                        feedbackObject.put("timestamp", timestamp.toString());
+                        if (!feedbackUpdated) {
+                        	skillName.put(feedbackObject);
+                        }
+
+                        languageName.put(skill_name, skillName);
+                        groupName.put(language_name, languageName);
+                        modelName.put(group_name, groupName);
+                        feedbackSkill.put(model_name, modelName, true);
+                        result.put("accepted", true);
+                        result.put("message", "Skill feedback updated");
+                        result.put("feedback", skill_feedback);
+                        return new ServiceResponse(result);
+                    }
+                }
+            }
+
+            addToSkillRatingJSON(call);
+
+            languageName.put(skill_name, createFeedbackArray(email, skill_feedback));
+            groupName.put(language_name, languageName);
+            modelName.put(group_name, groupName);
+            feedbackSkill.put(model_name, modelName, true);
+            result.put("accepted", true);
+            result.put("message", "Skill feedback updated");
+            result.put("feedback", skill_feedback);
+            return new ServiceResponse(result);
+
+        } else {
+            throw new APIException(422, "Access token not given.");
+        }
+
+
+    }
+
+    /* Utility function*/
+    public JSONArray createFeedbackArray(String email, String skill_feedback) {
+        JSONArray skillName = new JSONArray();
+        JSONObject feedbackObject = new JSONObject();
+        Timestamp timestamp = new Timestamp(System.currentTimeMillis());
+        feedbackObject.put("email", email);
+        feedbackObject.put("feedback", skill_feedback);
+        feedbackObject.put("timestamp", timestamp.toString());
+        skillName.put(feedbackObject);
+
+        return skillName;
+    }
+
+
+    public JSONObject createRatingObject() {
```

Comment:
This method is a dupe since it's already available in the `RateSkillService`. It's probably a better idea to remove this method and replace it with the call to that.

Reply:
The function in `RateSkillService` is a bit different from it. Also, `RateSkillService` is not used anymore since, thumbs up/down has been removed.I have made change in the function name to explain its use clearly.

Reply:
Have changed the code entirely and refactored it. It now works for all cases. Earlier, it missed few cases like not working for all Groups.